test/integration/RevShareContractsUpgraderIntegration.t.sol

```diff
@@ -0,0 +1,208 @@
+    uint32 internal constant INK_WITHDRAWAL_GAS_LIMIT = 1000000;
+    address internal constant INK_CHAIN_FEES_RECIPIENT = 0x0000000000000000000000000000000000000002;
+
+    bool internal constant IS_SIMULATE = true;
+
+    function setUp() public {
+        // Create forks for L1 (mainnet) and L2 (OP Mainnet)
+        _mainnetForkId = vm.createFork("http://127.0.0.1:8545");
+        _opMainnetForkId = vm.createFork("http://127.0.0.1:9545");
+        _inkMainnetForkId = vm.createFork("http://127.0.0.1:9546");
+
+        // Deploy contracts on L1
+        vm.selectFork(_mainnetForkId);
+
+        // Deploy RevShareContractsUpgrader and etch at predetermined address
+        revShareUpgrader = new RevShareContractsUpgrader();
+        vm.etch(REV_SHARE_UPGRADER_ADDRESS, address(revShareUpgrader).code);
+        revShareUpgrader = RevShareContractsUpgrader(REV_SHARE_UPGRADER_ADDRESS);
+
+        // Deploy RevShareUpgradeAndSetup task
+        revShareTask = new RevShareUpgradeAndSetup();
+    }
+
+    /// @notice Test the integration of upgradeAndSetupRevShare
+    function test_upgradeAndSetupRevShare_integration() public {
+        // Step 1: Record logs for L1→L2 message replay
+        vm.recordLogs();
+
+        // Step 2: Execute task simulation
+        revShareTask.simulate("test/tasks/example/eth/016-revshare-upgrade-and-setup/config.toml");
+
+        // Step 3: Relay deposit transactions from L1 to all L2s
+        uint256[] memory forkIds = new uint256[](2);
+        forkIds[0] = _opMainnetForkId;
+        forkIds[1] = _inkMainnetForkId;
+
+        address[] memory portals = new address[](2);
+        portals[0] = OP_MAINNET_PORTAL;
+        portals[1] = INK_MAINNET_PORTAL;
+
+        _relayAllMessages(forkIds, IS_SIMULATE, portals);
+
+        // Step 4: Assert the state of the OP Mainnet contracts
+        vm.selectFork(_opMainnetForkId);
+        _assertL2State(
+            OP_L1_WITHDRAWER,
+            OP_REV_SHARE_CALCULATOR,
+            OP_MIN_WITHDRAWAL_AMOUNT,
+            OP_L1_WITHDRAWAL_RECIPIENT,
+            OP_WITHDRAWAL_GAS_LIMIT,
+            OP_CHAIN_FEES_RECIPIENT
+        );
+
+        // Step 5: Assert the state of the Ink Mainnet contracts
+        vm.selectFork(_inkMainnetForkId);
+        _assertL2State(
+            INK_L1_WITHDRAWER,
+            INK_REV_SHARE_CALCULATOR,
+            INK_MIN_WITHDRAWAL_AMOUNT,
+            INK_L1_WITHDRAWAL_RECIPIENT,
+            INK_WITHDRAWAL_GAS_LIMIT,
+            INK_CHAIN_FEES_RECIPIENT
+        );
+    }
+
+    /// @notice Assert the state of all L2 contracts after upgrade
+    /// @param _minWithdrawalAmount Expected minimum withdrawal amount for L1Withdrawer
+    /// @param _l1Recipient Expected recipient address for L1Withdrawer
+    /// @param _gasLimit Expected gas limit for L1Withdrawer
+    /// @param _chainFeesRecipient Expected chain fees recipient (remainder recipient)
+    function _assertL2State(
```

Comment:
A couple additional things I'd like to see asserted: 
1. the versions of all the contracts are up to date with the new implementations
2. an actual withdrawal test to confirm that everything is wired together properly, and that the withdrawal happens as expected.

Reply:
Also in a new PR.